### PR TITLE
ZENKO-2803 Create Bucket

### PR DESCRIPTION
**Pull request by nicolas2bert**
Screen Shot 2020-09-17 at 11 10 53 AM

### CODE REVIEW

**Review by naren-rajendran — APPROVED** [review]

File: src/react/databrowser/buckets/BucketCreate.jsx [hunk 1]
```diff
@@ -78,50 +87,61 @@ function BucketCreate(props) {
...
-            <Input
+    const selectLocations = useMemo(() => {
+        return locationWithIngestion(locations, capabilities);
+    }, [locations, capabilities]);
+
+    return <Form innerRef={formRef}>
+        <F.Title> create new bucket </F.Title>
+        <F.Fieldset>
+            <F.Label tooltipMessages={['Must be unique']}>
+                Name
```
Comment (terrydervaux) [inline src/react/databrowser/buckets/BucketCreate.jsx]:
```suggestion
                Bucket Name
```

File: src/react/databrowser/buckets/BucketCreate.jsx [hunk 2]
```diff
@@ -78,50 +87,61 @@ function BucketCreate(props) {
...
-                value={bucket.name}
+                innerRef={register}
+                onChange={clearServerError}
                 autoComplete='off' />
-        </fieldset>
-        <fieldset>
+            <F.ErrorInput id='error-name' hasError={errors.name}> {errors.name?.message} </F.ErrorInput>
+        </F.Fieldset>
+        <F.Fieldset>
             <label htmlFor='locationConstraint'> Location Constraint </label>
```
Comment (terrydervaux) [inline src/react/databrowser/buckets/BucketCreate.jsx]:
```suggestion
            <label htmlFor='locationConstraint'> Select Storage Location </label>
```

File: src/react/databrowser/buckets/BucketCreate.jsx [hunk 3]
```diff
@@ -78,50 +87,61 @@ function BucketCreate(props) {
...
-            <label htmlFor='userName'> Name </label>
-            <Input
+    const selectLocations = useMemo(() => {
+        return locationWithIngestion(locations, capabilities);
+    }, [locations, capabilities]);
+
+    return <Form innerRef={formRef}>
+        <F.Title> create new bucket </F.Title>
+        <F.Fieldset>
+            <F.Label tooltipMessages={['Must be unique']}>
```
Comment (terrydervaux) [inline src/react/databrowser/buckets/BucketCreate.jsx]:
```suggestion
            <F.Label tooltipMessages={['The Bucket name cannot be changed after creation']}>
```

File: src/react/databrowser/buckets/BucketCreate.jsx [hunk 4]
```diff
@@ -78,50 +87,61 @@ function BucketCreate(props) {
...
-                onChange={onChange}
-                value={bucket.name}
+                innerRef={register}
+                onChange={clearServerError}
                 autoComplete='off' />
-        </fieldset>
-        <fieldset>
+            <F.ErrorInput id='error-name' hasError={errors.name}> {errors.name?.message} </F.ErrorInput>
+        </F.Fieldset>
+        <F.Fieldset>
```
Comment (terrydervaux) [inline src/react/databrowser/buckets/BucketCreate.jsx:109]: Suggestion: you can add a tooltip with the following content `The Storage Location cannot be changed after creation`

File: src/react/databrowser/buckets/BucketCreate.jsx [hunk 5]
```diff
@@ -78,50 +87,61 @@ function BucketCreate(props) {
...
-        <div className='sc-title'> create  bucket </div>
-        <fieldset>
-            <label htmlFor='userName'> Name </label>
-            <Input
+    const selectLocations = useMemo(() => {
+        return locationWithIngestion(locations, capabilities);
+    }, [locations, capabilities]);
+
+    return <Form innerRef={formRef}>
+        <F.Title> create new bucket </F.Title>
```
Comment (terrydervaux) [inline src/react/databrowser/buckets/BucketCreate.jsx]:
```suggestion
        <F.Title> create a new bucket </F.Title>
```

**Review by terrydervaux — APPROVED** [review]
LGTM 👍 

Nice to have:
- The location constraint shouldn't be pre-selected because the user can forget to select the right one, and storage location can't be changed after the creation.
- Use the same color for the background and the form component -> #121212(background)
- Add a line below the `Bucket Name` field with the following content `The Bucket Name should be unique`

**Comment by nicolas2bert** [comment]
> The location constraint shouldn't be pre-selected because the user can forget to select the right one, and storage location can't be changed after the creation.

I think it's ok to have a preselected location since the user has the ability to delete its bucket and create a new one.

> Use the same color for the background and the form component -> #121212(background)

For now, I think it is ok to delimit our forms using a different background.


> Add a line below the Bucket Name field with the following content The Bucket Name should be unique

I am using a tooltip for that.

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**